File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/FlowLayoutPanel/Functionality/FlowDirectionPage.vb
Namespace CompanionKit.Controls.FlowLayoutPanel.Functionality

	Public Class FlowDirectionPage

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
			InitializeComponent()

        End Sub

        ''' <summary>
        ''' Handles the SelectedIndexChanged event of the mobjFlowDirectionComboBox control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        Private Sub mobjFlowDirectionComboBox_SelectedIndexChanged(sender As Object, e As EventArgs) Handles mobjFlowDirectionComboBox.SelectedIndexChanged
            Select Case mobjFlowDirectionComboBox.SelectedIndex
                Case 0
                    mobjFlowLayoutPanel.FlowDirection = FlowDirection.LeftToRight
                    Exit Select
                Case 1
                    mobjFlowLayoutPanel.FlowDirection = FlowDirection.TopDown
                    Exit Select
                Case 2
                    mobjFlowLayoutPanel.FlowDirection = FlowDirection.RightToLeft
                    Exit Select
                Case 3
                    mobjFlowLayoutPanel.FlowDirection = FlowDirection.BottomUp
                    Exit Select
            End Select
        End Sub

        ''' <summary>
        ''' Handles the CheckedChanged event of the mobjWrapCheckBox control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        Private Sub mobjWrapCheckBox_CheckedChanged(sender As Object, e As EventArgs) Handles mobjWrapCheckBox.CheckedChanged
            mobjFlowLayoutPanel.WrapContents = mobjWrapCheckBox.Checked
        End Sub

        ''' <summary>
        ''' Handles the Load event of the FlowDirectÂionPage control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        Private Sub FlowDirectionPage_Load(sender As Object, e As EventArgs) Handles Me.Load
            mobjFlowDirectionComboBox.SelectedIndex = 0
            mobjWrapCheckBox.Checked = mobjFlowLayoutPanel.WrapContents
        End Sub

	End Class

End Namespace

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/FlowLayoutPanel/Functionality/FlowDirectionPage.cs
#region Using

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Forms;

#endregion

namespace CompanionKit.Controls.FlowLayoutPanel.Functionality
{
    public partial class FlowDirectionPage : UserControl
    {
        public FlowDirectionPage()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Handles the SelectedIndexChanged event of the mobjFlowDirectionComboBox control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjFlowDirectionComboBox_SelectedIndexChanged(object sender, EventArgs e)
        {
            switch (mobjFlowDirectionComboBox.SelectedIndex)
            {
                case 0:
                    mobjFlowLayoutPanel.FlowDirection = FlowDirection.LeftToRight; 
                    break;
                case 1:
                    mobjFlowLayoutPanel.FlowDirection = FlowDirection.TopDown;
                    break;
                case 2:
                    mobjFlowLayoutPanel.FlowDirection = FlowDirection.RightToLeft;
                    break;
                case 3:
                    mobjFlowLayoutPanel.FlowDirection = FlowDirection.BottomUp;
                    break;
            }
        }

        /// <summary>
        /// Handles the CheckedChanged event of the mobjWrapCheckBox control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjWrapCheckBox_CheckedChanged(object sender, EventArgs e)
        {
            mobjFlowLayoutPanel.WrapContents = mobjWrapCheckBox.Checked;
        }

        /// <summary>
        /// Handles the Load event of the FlowDirectÂionPage control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void FlowDirectionPage_Load(object sender, EventArgs e)
        {
            mobjFlowDirectionComboBox.SelectedIndex = 0;
            mobjWrapCheckBox.Checked = mobjFlowLayoutPanel.WrapContents;
        }
    }
}